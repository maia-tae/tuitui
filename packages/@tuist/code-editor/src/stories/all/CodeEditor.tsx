import { CodeEditor as component, CodeEditorProps as Props } from '../..'
import { config, Stories } from '../helper'

export const codeEditor: Stories<Props> = {
  name: 'CodeEditor',
  config: config,
  stories: [
    {
      name: 'simple',
      component,
      props: ctx => ({
        id: 'foo',
        source: `const foo = 'bar'`,
        lang: 'ts',
        onChange() {
          console.log('save')
        },
      }),
    },
  ],
}
