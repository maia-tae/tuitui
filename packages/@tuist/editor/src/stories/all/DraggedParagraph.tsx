import {
  DraggedParagraphProps as Props,
  DraggedParagraph as component,
} from '../..'
import { Stories, config } from '../helpers'

import { reference } from '@tuist/build'
import styled from 'styled-components'

const wrapper = styled.div`
  padding: 8px;
  background: white;
`

export const draggedParagraph: Stories<Props> = {
  name: 'Dragged paragraph',
  component,
  config,
  wrapper,
  stories: [
    {
      name: 'Long text',
      props(ctx) {
        const holder = ctx.state.test.e4
        const ids = Object.keys(holder.composition.g)
        const sourceId = ids[ids.length - 1]
        return {
          sourceRef: reference(holder),
          sourceId,
          sourceCompId: 'foo',
        }
      },
    },

    {
      name: 'Custom',
      props(ctx) {
        return {
          sourceRef: reference(ctx.state.test.e1),
          sourceId: 'bar',
          sourceCompId: 'foo',
        }
      },
    },
  ],
}
