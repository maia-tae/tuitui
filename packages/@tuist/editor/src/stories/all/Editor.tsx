import * as React from 'react'

import { EditorProps as Props, Editor as component } from '../..'
import { Stories, config, wrapper } from '../helpers'

import styled from 'styled-components'

const SomeChild = styled.div`
  border: 1px solid #333;
  background: #aaa;
  height: 80px;
  padding: 20px;
  color: #eee;
`

export const editorStories: Stories<Props> = {
  name: 'Editor',
  component,
  config,
  wrapper,
  stories: [
    {
      name: 'with title',
      props(ctx) {
        return { holder: ctx.state.test.e1 }
      },
    },

    {
      name: 'with title children',
      state: {
        test: {
          showTitleChildren: true,
        },
      },
      props(ctx) {
        return { holder: ctx.state.test.e1 }
      },
    },

    {
      name: 'with children',
      props(ctx) {
        return { holder: ctx.state.test.e1 }
      },
      children: [<SomeChild key="1">This is a child</SomeChild>],
    },

    {
      name: 'no composition',
      props(ctx) {
        return {
          holder: ctx.state.test.e5,
          titlePlaceholder: 'New Composition',
        }
      },
    },

    {
      name: 'no composition and no title',
      props(ctx) {
        return {
          holder: ctx.state.test.e6,
          titlePlaceholder: 'New Composition',
        }
      },
    },

    {
      name: 'title true',
      props(ctx) {
        return { holder: ctx.state.test.e2 }
      },
    },

    {
      name: 'no title',
      props(ctx) {
        return { holder: ctx.state.test.e3 }
      },
    },

    {
      name: 'list',
      props(ctx) {
        return { holder: ctx.state.test.e7 }
      },
    },

    {
      name: 'many pages',
      props(ctx) {
        return { holder: ctx.state.test.e4 }
      },
    },

    {
      name: 'readonly',
      props(ctx) {
        return { holder: ctx.state.test.e4, readonly: true }
      },
    },
  ],
}
