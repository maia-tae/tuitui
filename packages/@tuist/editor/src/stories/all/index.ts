//export * from './DraggedParagraph'
//export * from './Toolbox'
// export * from './Editor'
export * from './Columns'
//export * from './CustomParagraph'
