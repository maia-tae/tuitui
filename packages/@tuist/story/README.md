# @tuist/story

[![NPM version][npm-image]][npm-url]
[![Build status][travis-image]][travis-url]

[npm-image]: https://img.shields.io/npm/v/@tuist/story.svg?style=flat
[npm-url]: https://npmjs.org/package/@tuist/story
[travis-image]: https://img.shields.io/travis/tuist-org/tuist.svg?style=flat
[travis-url]: https://travis-ci.org/tuist-org/tuist

A simple storybook based on parcel to test components in an Overmind-React app.
