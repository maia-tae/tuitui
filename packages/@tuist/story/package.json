{
  "name": "@tuist/story",
  "version": "0.0.0",
  "description": "Simple component story book.",
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/tuist-org/tuist.git"
  },
  "author": "Maia Taewana <maia@tuitui.st>",
  "license": "MIT",
  "private": false,
  "scripts": {
    "build": "tsc -p setup/tsconfig.${NODE_ENV:=development}.json",
    "postbuild": "cp package.json ../../node_modules/@tuist/story",
    "clean": "rimraf .cache dist coverage ../../node_modules/@tuist/story",
    "prepare": "rimraf dist && cp -r ../../node_modules/@tuist/story/dist dist",
    "stories": "parcel serve --port 8888 --no-autoinstall src/stories/index.html",
    "test": "jest -c setup/jest.js",
    "test:watch": "jest -c setup/jest.js --watch --coverage false"
  },
  "dependencies": {
    "@tuist/build": "next",
    "@tuist/theme": "next",
    "jest-styled-components": "^7.0.3",
    "parcel": "^1.12.4"
  },
  "devDependencies": {}
}