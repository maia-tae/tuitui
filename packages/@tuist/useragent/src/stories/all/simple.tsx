import * as React from 'react'

import { Comp, styled, useOvermind } from '../helper'
import { Stories, config } from '../helper'

import { Icon } from '@tuist/styled'

const Wrapper = styled.div`
  padding: 5px;
`

const OnlineWidget: Comp<{}> = () => {
  const ctx = useOvermind()
  const { online } = ctx.state.useragent
  const name = online ? 'Online' : 'Offline'
  return (
    <Wrapper>
      <Icon icon={name} className={name} />
      {name}
    </Wrapper>
  )
}

const MessageWidget: Comp<{}> = () => {
  const ctx = useOvermind()
  // The message is updated through a hook
  const { message } = ctx.state.test
  return <Wrapper>{message}</Wrapper>
}

export const simpleStories: Stories<{}> = {
  name: 'Show state',
  component: OnlineWidget,
  config,
  stories: [
    {
      name: 'state changes',
    },
  ],
}

export const hookStories: Stories<{}> = {
  name: 'Triggers hook',
  component: MessageWidget,
  config,
  stories: [
    {
      name: 'changes on hook',
    },
  ],
}
