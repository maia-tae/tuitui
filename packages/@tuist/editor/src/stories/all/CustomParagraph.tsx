import { EditorProps as Props, Editor as component } from '../..'
import { config, makeComposition } from '../helpers'

import { Stories } from '@tuist/story'

export const customStories: Stories<Props> = {
  name: 'Custom paragraph',
  component,
  config,
  stories: [
    {
      name: 'simple',
      props(app: any) {
        return { holder: app.state.c1, id: 'c1' }
      },
      state: {
        c1: makeComposition({ id: 'para1', title: 'Custom paragraph' })
          .addParagraph('para2', {
            t: 'p',
            c: 'X',
            p: 1,
          })
          .setData('para2', { value: 0.5 })
          .done(),
      },
    },
  ],
}
