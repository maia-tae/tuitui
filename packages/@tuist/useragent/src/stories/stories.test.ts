import * as stories from './all'

import { testStories } from '@tuist/story'

testStories({}, stories)
