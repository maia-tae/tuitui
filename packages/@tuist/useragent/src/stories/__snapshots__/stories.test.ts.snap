// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show state should render state changes 1`] = `
<div
  className="theme-Show-state-state-changes"
>
  <div
    className="sc-hKFyIo muukU"
  >
    <div
      className="sc-eCApGN doPWto"
    >
      <div
        className="sc-jSFkmK euWlfV"
      >
        <div
          className="sc-gtssRu QCClT"
          style={
            Object {
              "cursor": "inherit",
            }
          }
        >
          state changes
        </div>
        <div
          className="sc-bdnylx gcNLkm"
        >
          &lt;OnlineWidget /&gt;
          <div
            className="jsx"
          >
            &lt;OnlineWidget /&gt;
          </div>
        </div>
      </div>
      <div
        className="sc-dlnjPT cbTXiB"
      >
        <div
          className="sc-fbIXFq dVEzPm"
        >
          <span
            className="Icon sc-jrsJCI faRQEi Offline"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-circle fa-w-16 "
              data-icon="circle"
              data-prefix="far"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </span>
          Offline
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Triggers hook should render changes on hook 1`] = `
<div
  className="theme-Triggers-hook-changes-on-hook"
>
  <div
    className="sc-hKFyIo muukU"
  >
    <div
      className="sc-eCApGN doPWto"
    >
      <div
        className="sc-jSFkmK euWlfV"
      >
        <div
          className="sc-gtssRu QCClT"
          style={
            Object {
              "cursor": "inherit",
            }
          }
        >
          changes on hook
        </div>
        <div
          className="sc-bdnylx gcNLkm"
        >
          &lt;MessageWidget /&gt;
          <div
            className="jsx"
          >
            &lt;MessageWidget /&gt;
          </div>
        </div>
      </div>
      <div
        className="sc-dlnjPT cbTXiB"
      >
        <div
          className="sc-fbIXFq dVEzPm"
        >
          no message
        </div>
      </div>
    </div>
  </div>
</div>
`;
